docs/src/main/asciidoc/kafka-streams-guide.adoc

```diff
@@ -0,0 +1,1141 @@
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=kafka-streams-quickstart-aggregator \
+    -Dextensions="kafka-streams,resteasy-jsonb" \
+    && cd ..
+----
+
+This creates the `aggregator, project with the Quarkus extension for Kafka Streams and with RESTEasy support for JSON-B.
+
+=== The Pipeline Implementation
+
+Let's begin the implementation of the stream processing application by creating
+a few value objects for representing temperature measurements, weather stations and for keeping track of aggregated values.
+
+First, create the file `aggregator/src/main/java/org/acme/quarkus/sample/kafkastreams/model/WeatherStation.java`,
+representing a weather station, with the following content:
+
+[source, java]
+----
+package org.acme.quarkus.sample.kafkastreams.model;
+
+import io.quarkus.runtime.annotations.RegisterForReflection;
+
+@RegisterForReflection
+public class WeatherStation {
+
+    public int id;
+    public String name;
+}
+----
+
+Then the file `aggregator/src/main/java/org/acme/quarkus/sample/kafkastreams/model/TemperatureMeasurement.java`,
+representing temperature measurements for a given station:
+
+[source, java]
+----
+package org.acme.quarkus.sample.kafkastreams.model;
+
+import java.time.Instant;
+
+public class TemperatureMeasurement {
+
+    public int stationId;
+    public String stationName;
+    public Instant timestamp;
+    public double value;
+
+    public TemperatureMeasurement(int stationId, String stationName, Instant timestamp, double value) {
+        this.stationId = stationId;
+        this.stationName = stationName;
+        this.timestamp = timestamp;
+        this.value = value;
+    }
+}
+----
+
+And finally `aggregator/src/main/java/org/acme/quarkus/sample/kafkastreams/model/Aggregation.java`,
+which will be used to keep track of the aggregated values while the events are processed in the streaming pipeline:
+
+[source, java]
+----
+package org.acme.quarkus.sample.kafkastreams.model;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+import io.quarkus.runtime.annotations.RegisterForReflection;
+
+@RegisterForReflection
+public class Aggregation {
+
+    public int stationId;
+    public String stationName;
+    public double min = Double.MAX_VALUE;
+    public double max = Double.MIN_VALUE;
+    public int count;
+    public double sum;
+    public double avg;
+
+    public Aggregation updateFrom(TemperatureMeasurement measurement) {
+        stationId = measurement.stationId;
+        stationName = measurement.stationName;
+
+        count++;
+        sum += measurement.value;
+        avg = BigDecimal.valueOf(sum / count)
+                .setScale(1, RoundingMode.HALF_UP).doubleValue();
+
+        min = Math.min(min, measurement.value);
+        max = Math.max(max, measurement.value);
+
+        return this;
+    }
+}
+----
+
+All these types carry the `@RegisterForReflection`, allowing them to be instantiated reflectively when running the application in native mode.
```

Comment:
Except one :-)

Reply:
Actually, the reason is to instruct the Graal compiler to make sure the classes are not removed during the dead-code elimination

Reply:
The classes are referenced in the code, so they shouldn't be removed entirely. It's only the parameterless constructor that'll be removed. In fact, just re-tested in native mode, and it's only needed for `WeatherStation`, as that's the only type instantiated reflectively by a `Deserializer`. Pushed a commit for removing the superfluous ones and moving the sentence right to that listing itself.